datafusion/expr/src/expr_schema.rs

```diff
@@ -113,7 +114,22 @@ impl ExprSchemable for Expr {
                     .iter()
                     .map(|e| e.get_type(schema))
                     .collect::<Result<Vec<_>>>()?;
-                Ok(arg_data_types[0].clone())
+                let arg_data_type = arg_data_types[0].clone();
```

Comment:
We have type checking in

https://github.com/apache/arrow-datafusion/blob/0302d6530a5a7fffe7da40e0f3b13794a9c693ff/datafusion/sql/src/expr/function.rs#L349-L361

so I think we dont need the duplicated check here.